### PR TITLE
refactor(item): remove deprecated apis

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: Internal

---------

## What is the current behavior?

Item the following deprecated APIs:

**Props**

1. fill
2. shape
3. counter
4. counterFomatter

**Slots**

1. helper
2. error

## What is the new behavior?

- These properties and slots (including implementations) have been removed.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

## Other information

BREAKING CHANGE:

- The `helper` slot has been removed. Developers should use the `helperText` property on `ion-input` and `ion-textarea`.
- The `error` slot has been removed. Developers should use the `errorText` property on `ion-input` and `ion-textarea`.
- Counter functionality has been removed including the `counter` and `counterFormatter` properties. Developers should use the properties of the same name on `ion-input` and `ion-textarea`.
- The `fill` property has been removed. Developers should use the property of the same name on `ion-input`, `ion-select`, and `ion-textarea`.
- The `shape` property has been removed. Developers should use the property of the same name on `ion-input`, `ion-select`, and `ion-textarea`.

### CODE REVIEW

File: core/src/components/item/item.md.scss [hunk 1]
```diff
@@ -32,65 +32,22 @@
   text-transform: none;
 }
 
-:host(.item-fill-outline) {
-  --highlight-height: 2px;
-}
-
 // Item Fill: None
 // --------------------------------------------------
```
Comment (amandaejohnston) [inline core/src/components/item/item.md.scss:36]:
```suggestion
```
This comment is no longer relevant.

File: core/src/components/item/item.ios.scss [hunk 2]
```diff
@@ -204,18 +202,10 @@
...
 }
 
-
 // iOS Stacked & Floating Inputs
 // --------------------------------------------------
 
 :host(.item-label-floating),
 :host(.item-label-stacked) {
   --min-height: 68px;
 }
```
Comment (amandaejohnston) [inline core/src/components/item/item.ios.scss:211]: Was this missed? The corresponding styles in MD have been removed in this PR.

I was a little confused about why the `item-label-` styles are being removed in this PR, but it looks like the class hasn't been on item for a while anyway, so we're just removing the remaining styles now since it's convenient? Is that accurate?

File: core/src/components/item/item.md.scss [hunk 3]
```diff
@@ -32,65 +32,22 @@
...
   border-style: var(--border-style);
   border-color: var(--highlight-background);
 }
 
-:host(.item-fill-solid.item-interactive.ion-focus) .item-native,
-:host(.item-fill-solid.item-interactive.item-has-focus) .item-native,
-:host(.item-fill-solid.item-interactive.ion-touched.ion-invalid) .item-native {
+:host(.item-interactive.ion-focus) .item-native,
+:host(.item-interactive.item-has-focus) .item-native,
+:host(.item-interactive.ion-touched.ion-invalid) .item-native {
```
Comment (amandaejohnston) [inline core/src/components/item/item.md.scss:50]: Since these styles previously only applied on `fill="solid"`, wouldn't we want to remove them completely rather than have them apply in all cases?

File: core/src/components/item/item.scss [hunk 4]
```diff
@@ -412,7 +412,7 @@ button, a {
   box-sizing: border-box;
 }
 
-:host(.item-label-stacked),
```
Comment (amandaejohnston) [inline core/src/components/item/item.scss:415]: Why are we only removing the `stacked` style here? Other parts of the PR removed both.

File: core/src/components/item/item.scss [hunk 5]
```diff
@@ -521,35 +515,6 @@ button, a {
...
-:host(.item-interactive.ion-invalid) ::slotted([slot="error"]) {
-  display: block;
-}
-
-// Item Datetime
-// --------------------------------------------------
-
-:host(:not(.item-label)) ::slotted(ion-datetime) {
-  --padding-start: 0;
-}
```
Comment (amandaejohnston) [inline core/src/components/item/item.scss:543]: Why was this style removed? It doesn't look related to any of the deprecated APIs.

File: core/src/components/item/test/basic/item.e2e.ts [hunk 6]
```diff
@@ -26,8 +26,7 @@ configs().forEach(({ title, screenshot, config }) => {
         </style>
         <ion-list>
           <ion-item>
-            <ion-label>Item with helper</ion-label>
-            <div slot="helper">Helper</div>
+            <ion-label>Item</ion-label>
```
Comment (amandaejohnston) [inline core/src/components/item/test/basic/item.e2e.ts:29]: I think this item can be removed entirely; it doesn't seem to be adding anything to the test now.

**Review by amandaejohnston — CHANGES_REQUESTED** [review]

File: core/src/components/item/item.ios.scss (hunk 2 above)
Comment (liamdebeasi) [reply core/src/components/item/item.ios.scss:211]: No, these should stay. These have to do with `ion-label` position property. We may need to discuss if the floating/stacked/fixed variants are still needed on `ion-label`, but the behaviors were not deprecated originally so I kept them in.

File: core/src/components/item/item.md.scss (hunk 3 above)
Comment (liamdebeasi) [reply core/src/components/item/item.md.scss]: fixed in https://github.com/ionic-team/ionic-framework/pull/29102/commits/e65dbf6f9a06a001fae0031e44cfe05d063c6a13

File: core/src/components/item/item.scss (hunk 5 above)
Comment (liamdebeasi) [reply core/src/components/item/item.scss:543]: This is part of the form control + item logic. As part of this refactor we are removing the tight coupling between items and form controls.

File: core/src/components/textarea/textarea.scss [hunk 7]
```diff
@@ -116,10 +116,6 @@
   align-self: baseline;
 }
 
-:host-context(ion-item:not(.item-label)) {
-  --padding-start: 0;
-}
```
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply core/src/components/textarea/textarea.scss:121]: This only applied with the legacy textarea syntax. With the modern syntax the padding start is always 0.

File: core/src/components/item/item.scss (hunk 4 above)
Comment (liamdebeasi) [reply core/src/components/item/item.scss]: Fixed in https://github.com/ionic-team/ionic-framework/pull/29102/commits/1e480e7e3c45629b89587142766d926051a00d32

File: core/src/components/item/test/basic/item.e2e.ts (hunk 6 above)
Comment (liamdebeasi) [reply core/src/components/item/test/basic/item.e2e.ts:29]: I don't understand. This test verifies that safe area is being added properly, but I didn't remove any of that logic.

File: core/src/components/item/item.md.scss [hunk 8]
```diff
@@ -298,24 +203,6 @@
...
 }
 
-
-// Material Design Floating/Stacked Label
-// --------------------------------------------------
-
-:host(.item-label-stacked) ::slotted([slot="end"]),
-:host(.item-label-floating) ::slotted([slot="end"]) {
-  @include margin($item-md-label-slot-end-margin-top, $item-md-label-slot-end-margin-end, $item-md-label-slot-end-margin-bottom, $item-md-label-slot-end-margin-start);
-}
```
Comment (amandaejohnston) [inline core/src/components/item/item.md.scss:308]: Why were these removed if we're keeping the `ion-label` position styles?

File: core/src/components/item/item.md.scss [hunk 9]
```diff
@@ -348,14 +235,6 @@
 // }
 
 
-// Material Design Stacked & Floating Inputs
-// --------------------------------------------------
-
-:host(.item-label-floating),
-:host(.item-label-stacked) {
-  --min-height: 55px;
-}
```
Comment (amandaejohnston) [inline core/src/components/item/item.md.scss:357]: Why were these removed if we're keeping the `ion-label` position styles?

File: core/src/components/item/test/basic/item.e2e.ts (hunk 6 above)
Comment (amandaejohnston) [reply core/src/components/item/test/basic/item.e2e.ts:29]: There are now two items in these screenshots, but we only need one to test the safe area. Previously the modified item was checking how the `helper` slot handles safe area; now it's just a regular item, so the other item in the screenshots handles things by itself.

**Review by mapsandapps — CHANGES_REQUESTED** [review]
I think I found two other things that can be removed:

- The `item-md-input-counter-letter-spacing` variable in [item.md.vars.scss](https://github.com/ionic-team/ionic-framework/blob/1e480e7e3c45629b89587142766d926051a00d32/core/src/components/item/item.md.vars.scss#L72-L73)
- The `.item-bottom` styles in [item.scss](https://github.com/ionic-team/ionic-framework/blob/1e480e7e3c45629b89587142766d926051a00d32/core/src/components/item/item.scss#L309-L332)

File: core/src/components/item/item.md.scss (hunk 8 above)
Comment (liamdebeasi) [reply core/src/components/item/item.md.scss]: Good catch. I had originally removed this before I realized that I should not have. These instances were the ones I missed when reverting.

File: core/src/components/item/item.md.scss (hunk 9 above)
Comment (liamdebeasi) [reply core/src/components/item/item.md.scss]: Fixed in https://github.com/ionic-team/ionic-framework/pull/29102/commits/02e5958b825b3a07174b178fd29b80b34feec0be

**Comment by liamdebeasi** [comment]
Bottom styles and variables removed in https://github.com/ionic-team/ionic-framework/pull/29102/commits/d48224689337376c6cd38978c5675b61be44d0bb and https://github.com/ionic-team/ionic-framework/pull/29102/commits/f98e95ded4650f7b1680cf3eca2c53d0002d646c.

File: core/src/components/item/test/basic/item.e2e.ts (hunk 6 above)
Comment (liamdebeasi) [reply core/src/components/item/test/basic/item.e2e.ts:29]: Ahh ok I see. I can fix

Comment (liamdebeasi) [reply core/src/components/item/test/basic/item.e2e.ts]: Fixed https://github.com/ionic-team/ionic-framework/pull/29102/commits/cbdfdd677819440d28a9837a96cb34cf42bcfc50

**Review by mapsandapps — APPROVED** [review]
